bindings/rust/bench/src/rustls.rs

```diff
@@ -72,131 +68,157 @@ impl RustlsHarness {
     }
 }
 
-impl TlsBenchHarness for RustlsHarness {
-    fn new(
+/// Clients and servers have different config types in Rustls, so wrap them in an enum
+pub enum RustlsConfig {
+    Client(Arc<ClientConfig>),
+    Server(Arc<ServerConfig>),
+}
+
+impl TlsConnection for RustlsConnection {
+    type Config = RustlsConfig;
+
+    fn name() -> String {
```

Comment:
Why do we need this function?

Reply:
This is mainly for printing out the names in the criterion benches; it made the most sense to me for the impls to own the name instead of listing it out separate as a parameter in mutliple places.